pkg/cmd/authenticated_cli_command.go

```diff
@@ -175,14 +175,50 @@ func (c *AuthenticatedCLICommand) GetMetricsClient() (*ccloudv2.MetricsClient, e
 	return c.metricsClient, nil
 }
 
+func (c *AuthenticatedCLICommand) GetSchemaRegistryClientByFlags(cmd *cobra.Command, unsafeTrace bool) error {
```

Comment:
The regular `GetSchemaRegistryClient()` function should be able to read flags, right?

Reply:
Yes and no, i need to add a different branch where it's neither cloud or on-prem login. And this function is just i refactored some reusable code.

Reply:
> i need to add a different branch where it's neither cloud or on-prem login

This sounds good, we should add it to `GetSchemaRegistryClient()` so there's only one function!

Reply:
But that's good chunk of reusable code so i took it out